File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.java
public class Solution {
    public int solution(int n) {
        int answer = 1;
        int first = 1;
        int second = 2;

        if (n == 1 || n == 2) return answer;

        for (int i = 3; i <= n; i++) {
            int temp = (first + second) % 1000000007;
            first = second;
            second = temp;
        }

        answer = second;

        return answer;
    }
}

File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.js
function solution(n) {
    if (n === 1) return 1;
    if (n === 2) return 2;
    
    let [first, second] = [1, 2];
    
    for (let i = 3; i <= n; i++) {
        [first, second] = [second, (first + second) % 1000000007];
    }
        
    return second;
}